### PR TITLE
[CHAINSAW-67] Allow to read gzip files.

### PR DESCRIPTION
**Pull request by tzachi-dar**
Signed-off-by: Tzachi Dar <tzachi.dar@gmail.com>

### CODE REVIEW

File: src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java [hunk 1]
```diff
@@ -314,6 +315,10 @@ public void activateOptions() {
         private Reader reader;
         private FileObject fileObject;
 
+        private boolean IsGZip(String fileName) {
```
Comment (jvz) [inline src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java]:
```suggestion
        private boolean isGZip(String fileName) {
```

File: src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java [hunk 2]
```diff
@@ -403,8 +409,16 @@ public void run() {
...
+                            if(IsGZip(getFileURL())) {
+                                InputStream gzipStream = new GZIPInputStream(fileObject.getContent().getInputStream());
+                                Reader decoder = new InputStreamReader(gzipStream,  "UTF-8");
+                                BufferedReader bufferedReader = new BufferedReader(decoder);
+                                process(bufferedReader);
+                                readingFinished = true;
+                            }
                             //could have been truncated or appended to (don't do anything if same size)
-                            if (fileObject.getContent().getSize() < lastFileSize) {
+                             if (fileObject.getContent().getSize() < lastFileSize) {
```
Comment (jvz) [inline src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java]: Looks like an errant space here.

File: src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java [hunk 3]
```diff
@@ -403,8 +409,16 @@ public void run() {
                             } catch (Error err) {
                                 getLogger().info(getPath() + " - unable to refresh fileobject", err);
                             }
+
+                            if(IsGZip(getFileURL())) {
```
Comment (jvz) [inline src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java]:
```suggestion
                            if (isGZip(getFileURL())) {
```

File: src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java (hunk 2 above)
Comment (tzachi-dar) [reply src/main/java/org/apache/log4j/chainsaw/vfs/VFSLogFilePatternReceiver.java]: Fixed.

**Comment by tzachi-dar** [comment]
Opened issue.
https://issues.apache.org/jira/browse/CHAINSAW-67

Please review again.

**Pull request merged**